sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchServiceClient.java

```diff
@@ -295,23 +308,29 @@ public Indexer getIndexer(String indexerName) {
      * @param indexerName the name of the indexer to delete
      */
     public void deleteIndexer(String indexerName) {
-        deleteIndexerWithResponse(indexerName, null, null, Context.NONE);
+        deleteIndexerWithResponse(new Indexer().setName(indexerName), null, null, Context.NONE);
     }
 
     /**
      * Deletes an Azure Cognitive Search indexer.
      *
-     * @param indexerName the name of the indexer to delete
-     * @param accessCondition the condition where the operation will be performed if the ETag on the server matches or
+     * @param indexer the search {@link Indexer}
+     * @param onlyIfUnchanged the condition where the operation will be performed if the ETag on the server matches or
      * doesn't match specified values
      * @param requestOptions additional parameters for the operation. Contains the tracking ID sent with the request to
      * help with debugging
      * @param context the context
      * @return a response signalling completion.
      */
-    public Response<Void> deleteIndexerWithResponse(String indexerName, MatchConditions accessCondition,
+    public Response<Void> deleteIndexerWithResponse(Indexer indexer, Boolean onlyIfUnchanged,
```

Comment:
Not a huge fan to switching this to `Indexer` just to support simpler `ETag` matching. We switched the burden from creating a simple `MatchConditions` class to creating a `Indexer` class.

Reply:
Exactly a valid point.
@heaths do you have any comments on this?

Reply:
See https://azure.github.io/azure-sdk/general_design.html#conditional-requests

This is a pattern established in AppConfiguration and simplifies how developers use ETag. No one else seems to expose MatchConditions directly. But even if we did, in order to use the ETag of an index (or any model), you would've had to retrieved it. So you already have it, right?